### ISSUE TITLE
Will Docker CE for AWS be updated for Docker 18.06 (and beyond)?

### ISSUE THREAD

**Issue by kinghuang** [docker/for-aws#164]
### Expected behavior

Docker for AWS should be updated for Docker 18.06.0.

### Actual behavior

Docker for AWS has not been updated for either Docker 18.03.1 or Docker 18.06.0.

### Information

The Docker for AWS documentation has been restored (#163), but there's very little indication if Docker for AWS is going to receive any more updates, or if it is abandoned. The Docker for AWS templates have not been updated for Docker 18.03.1 or Docker 18.06.0, and the related Moby Linux AMIs have not been updated past Docker 18.04.0.

https://store.docker.com/editions/enterprise/docker-ee-aws states that Docker EE for AWS has been deprecated in favour of Docker Certified Infrastructure for AWS.

https://store.docker.com/editions/community/docker-ce-aws still does not state that Docker CE for AWS is deprecated.

### Steps to reproduce the behavior

  1. Visit https://docs.docker.com/docker-for-aws/ and deploy Docker CE for AWS.
  2. Observe that Docker 18.03.0 is deployed.

**Comment by kinghuang** [comment]
[Justin Cormack on the #docker-for-aws Slack channel said](https://dockercommunity.slack.com/archives/C2TJD5L5U/p1532375561000461):

> Moby Linux is not going to be maintained as it was a very early release of LinuxKit (it is all in the repo history if you dig around around two years ago...) There was work ongoing at one point to finish porting them to modern LinuxKit, not sure if there is anything from that that would be useful to open source.

[This twitter thread suggests the Docker for AWS templates are no more](https://twitter.com/hairyhenderson/status/1019959573854203905):

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">I touched on this in my DockerCon talk, that the templates are no more (small usage, not flexible enough) so the new tools use terraform and ansible plus documentation/guides.</p>&mdash; Bret Fisher (@BretFisher) <a href="https://twitter.com/BretFisher/status/1020401675184607233?ref_src=twsrc%5Etfw">July 20, 2018</a></blockquote>

**Comment by brawong** [comment]
Yes. We are working on building it... cc @joeabbey 
Will provide an update once it's ready. Thanks!

**Comment by vce-xx** [comment]
Any progress ? @brawong  @joeabbey

**Comment by kinghuang** [comment]
FWIW, there's AMIs now for Moby Linux 18.06.0 and 18.06.1 (at least in us-west-2), along with the corresponding docker4x images. I've been using them in my newest clusters.

**Comment by brawong** [comment]
Yes - I believe we had posted these back in late August. Thanks for checking in.

**Comment by kinghuang** [comment]
@brawong Will the CloudFormation templates be updated to match? They're still pointing to the 18.03.0 images.

**Comment by philipp-paland** [comment]
@kinghuang To use those newer versions is is enough to change the AMI ids under the key "AWSRegionArch2AMI" in the CloudFormation template? Or did you do any other changes?

What's your experience regarding stability?

**Comment by philipp-paland** [comment]
@brawong I don't see why you closed this issue, without an updated CF template it is not really resolved.

**Comment by brawong** [comment]
Right - re-opening until CF templates are updated. Thanks

**Comment by joeabbey** [comment]
Ah, they were placed in a staging bucket.

Just uploaded them from staging -> production

https://download.docker.com/aws/stable/Docker.tmpl
https://download.docker.com/azure/stable/Docker.tmpl

Please let me know if anything is missing there 👍

**Comment by hairyhenderson** [comment]
Thanks @joeabbey - I can confirm this is working how I expect:

```console
$  curl -s https://editions-us-east-1.s3.amazonaws.com/aws/stable/18.06.1/Docker-no-vpc.tmpl | grep 'Docker CE for AWS'
    "Description": "Docker CE for AWS 18.06.1-ce (18.06.1-ce-aws1)",
$ curl -s https://editions-us-east-1.s3.amazonaws.com/aws/stable/18.06.1/Docker.tmpl | grep 'Docker CE for AWS' 
    "Description": "Docker CE for AWS 18.06.1-ce (18.06.1-ce-aws1)",
```

🚀

**Comment by joeabbey** [comment]
Thanks for confirmation!  Closing this out. 

Update for 18.09 should go much more smoothly.

**Comment by hairyhenderson** [comment]
@joeabbey will the 18.09 D4AWS update be announced anywhere, or should we keep guessing URLs and curling them periodically? 😉

**Comment by mrmachine** [comment]
How about some release notes? Docs still say the latest version is 18.03

**Comment by LennyPenny** [comment]
@hairyhenderson do you have any news for docker 18.09 on ami linux?

**Comment by hairyhenderson** [comment]
@LennyPenny as far as I know, the guess-and-curl approach is still the one to use 🤷‍♂️. I'm not a Docker employee, so I have no inside knowledge of the release processes...

**Comment by kinghuang** [comment]
With the lack of updates on the CloudFormation template and the Cloudstor plugin, I think Docker for AWS is basically dead. I've switched over to using my own Terraform template to stand up Swarm clusters on AWS.

Perhaps it's time to band together as a community and create a replacement for D4AWS?

**Comment by tomalok** [comment]
I'm not holding my breath for a new docker4x release.  From what I've gathered (i.e. "not much") it's been replaced by some kind of an Enterprise solution.  The CloudFormation templates left quite a bit to be desired (which is why I released various template tweaks https://gitlab.com/tomalok/docker4aws).

My primary concern is the continued availability and development of the CloudStor plugin.  While it's true that you *CAN* use the last CloudStor plugin (18.06.1-ce-aws1) with the latest docker (18.09.1-ce), I've not idea how long compatibility will last.  As it is today, CloudStor is not compatible with instances that use NVMe for EBS volumes, so I'm forced to use REX-Ray for that, but CloudStor's EFS support still surpasses REX-Ray's, IMHO.
